src/components/view/view-event-list.tsx

```diff
@@ -420,6 +440,18 @@ const ExchangeRow = observer(({
         <PathAndQuery title={ request.parsedUrl.pathname + request.parsedUrl.search }>
             { request.parsedUrl.pathname + request.parsedUrl.search }
         </PathAndQuery>
+        <Duration>
+            {durationMs > 0
+                ? (
+                    (durationMs < 100) ?
+                        (sigFig(durationMs, 2) + 'ms') :
+                        (durationMs < 1000 ? sigFig(durationMs, 1) + 'ms' : (durationMs < 10000 ? sigFig(durationMs / 1000, 3) + ' s' : sigFig(durationMs / 1000, 1) + ' s'))
+                )
```

Comment:
I think this whole chunk would probably be nicer as a reusable function like `formatDuration` somewhere, e.g. src/util/index.ts.